### ISSUE TITLE
Rename project to kiwix-js?

### ISSUE THREAD

**Issue by kelson42** [kiwix/kiwix-js#248]
@Jaifroid suggested me the idea and I think that:
* This better represents what it is
* This is easier to say
* This sounds better

What do you think?

**Comment by Jaifroid** [comment]
The main thing is to have a name that distinguishes the x86 / x64 versions from those versions written in JavaScript, which was my reason for suggesting Kiwix JS as a concise name for the Windows UWP version. For naïve users, HTML5 or JS won't mean anything, but they'll see (or they'll be told in the app description) that there are two "editions" of Kiwix, a full-featured one and a "light" one (HTML5 / JS) especially suited to tablets and other mobile devices that can't run the full-featured executable.

What the app is called is probably only relevant in a store context, but it does become quite important in such a context. Especially with Windows 10 which has a very large install base, even though a lot of users (businesses in particular) are barely aware of the Store at the moment. I believe that will change as Microsoft and other vendors are looking to distribute x64 software via the Store (e.g., Office). There is also Microsoft's upcoming "Surface Laptop" (MacBook Air competitor), shipping with a "Windows 10 S" edition (S for Store) which ONLY allows software to be installed from the Store. (Users can upgrade to Windows 10 Pro for free, but many students who will use the thing basically for browsing, Office and Spotify, will never bother.) 

So, in a Store context, the "light" version of Kiwix needs a simple, snappy name that distinguishes it immediately and visually from the full-featured version. Kiwix JS seemed the most concise way of expressing it, and perhaps less "nerdy" than Kiwix HTML5 for the naïve user.

Whatever it's called, it would be good to bring into the HTML5 version the ability to download and save a ZIM file from the repository, but that's another issue: #114 (Add a way to easily download ZIM archives).

**Comment by mossroy** [comment]
I agree with you.
On the other hand, if #116 is implemented one day, it will become mostly wasm (WebAssembly) code instead of js code. But, well, I don't find a better name.

If we decide to do it, there are few places where it has to be renamed.
And I think it would be better not to rename it in manifest.json because it's in the id of the Firefox extension : kiwix-html5-unlisted@kiwix.org (or kiwix-html5-listed@kiwix.org for the released versions). If we change it, I'm afraid Firefox will not upgrade existing versions of the extension.

**Comment by Jaifroid** [comment]
I don't think there's any rush to change things in this repository or on FF. But if @kelson42 agrees, I'd like to post the UWP app in the Windows Store as "Kiwix JS" (possibly initially as "Kiwix JS Beta" while ironing out bugs). We've reserved both names in the store: "Kiwix" and "Kiwix JS". It's easy to reserve other names if someone comes up with something better.

**Comment by Jaifroid** [comment]
@kelson42 @Mossroy  What do you think of "Kiwix Mobile" as the Store name for the Windows app? Microsoft has "Word Mobile" in the Store, and it is installable on tablets and laptops as well as phones.

**Comment by Popolechien** [comment]
@Jaifroid I'm a very naive user and "mobile" for me would definitely describe the full app running on a phone/tablet. Why not simply call it Kiwix Light (or Lite)? It's the most descriptive and unambiguous name I can think of. Kiwix extension or Kiwix S could also work, but I don't know how well that fits Win10 naming conventions.

**Comment by kelson42** [comment]
@Jaifroid I'm not a supporter of "Kiwix Mobile" as it might be confusing with other mobile versions of Kiwix for Android and iOS.

So, to wrap-up, it seems that we agree that "Kiwix JS" is a better name for what we are doing. That is the most important. I propose (sorry if I'm a bit bold):
* "Kiwix JS" (be careful of the case sensitivity) will be the name of all current kiwix-html5 products, so extensions and Windows UWP version.
* I will change the git repos that way from "kiwix-html5" to "kiwix-js" and "kiwix-html5-windows" to "kiwix-js-windows"
* Changes in kiwix-js (except web extensions shop identifiers) will be changed then (I will open a separate ticket)
* Web extensions shop descriptions/title will be changed at the next release
* Web extensions ids change should be handled in an other ticket (separately) to check if that work at all.

**Comment by kelson42** [comment]
@Popolechien "lite" or "light" are OK for me too... but it's not really descriptive, that is why I prefer "JS".... What do other participants think about that proposal?

**Comment by Popolechien** [comment]
@kelson42 If you're trying to convey that the product is ligther than the "real thing" I would stick to S as in Small (there's also a good precedent with iPhone 5/5S so we're not breaking new ground, even if it's on Windows and not iOS). 

"JS" sounds like there's JavaScript in it, which is fine (and true) but that conveys code, not a marked product difference. For an average joe like myself, code is something I do not understand and therefore have to worry about, not embrace.

**Comment by Jaifroid** [comment]
> So, to wrap-up, it seems that we agree that "Kiwix JS" is a better name for what we are doing. That is the most important. I propose (sorry if I'm a bit bold):
> * "Kiwix JS" (be careful of the case sensitivity) will be the name of all current kiwix-html5 products, so extensions and Windows UWP version.
> * I will change the git repos that way from "kiwix-html5" to "kiwix-js" and "kiwix-html5-windows" to "kiwix-js-windows"
> * Changes in kiwix-js (except web extensions shop identifiers) will be changed then (I will open a separate ticket)
> * Web extensions shop descriptions/title will be changed at the next release
> * Web extensions ids change should be handled in an other ticket (separately) to check if that work at all. 

I'm perfectly happy with this. (I don't particularly like the "Light" label on products, and particularly not "Lite" which is horrid commercialese -- it reminds me of some dieting aid...)

**Comment by mossroy** [comment]
Regarding the github repos (and all the technical stuff), kiwix-js looks fine for me.
For distribution through stores, the displayed name needs to be simple for the end-user (even if the hidden id is kiwix-js, or anything else).
When there is no other version of Kiwix, I think we could keep a display name "Kiwix" : if there's no other choice, the end-user doesn't care about the underlying technology.
When there is another version of Kiwix (on desktop Windows 10, for example), we need a way for the end-user to understand it's not the fully-featured version. S, Lite or Light are OK for me. If it's only a display name, we can change it later if necessary (who knows : one day, this js version might have all the features, be fast and fully compatible, and thus would not be a "light" version any more ;-) )

To sum up, I would vote to use kiwix-js everywhere, except in the display name for end-users. Where I would use "Kiwix" when there is no other version, or "Kiwix with-any-suffix" when it might be confusing with the original Kiwix version.

**Comment by sharun-s** [comment]
Agree with @mossroy said above. Will just add last month on the #218 thread we had a somewhat related discussion on the output of this repo.
 
It came up that the code already has many bits and pieces that allow it to be used in 5 different "modes"

- Standalone 
distribute zims and code folder on a usb key/sdcard with instructions to click the index.html file to run [This is how I plan to use this code btw as it requires least instructions to anyone with low computer literacy i.e. most people who are offline]
- Local server based
which would make duplications of massive zims across many clients unnecessary for local classroom type scenarios
- WebExtension
- WebApp (phone gap/corova based) on app store
- NativeApp eg the Firefox OS app

We decided focus of the repo would be on webextension during that thread (along with FFOS specific code which mossroy is still using). Cordova/evopedia specific stuff was removed. @Jaifroid has shown it can be used as the basis of a UWP app. It can also obviously be the basis of many future modes such as a Node module, REST API, electron app, in rasp pi/amazon echo/smart tv's whatever.  

So while the current code serves as upstream for other platform/"mode" specific implementations it also holds/will hold "mode" specific code in HEAD or history. Ideally the output of this repo is just one performant mode/implementation/UI agnostic kiwix.js library. It would be responsible for read and search of ZIM files. As the app/extension/mode specific parts mature they can be pushed into separate separate repos with their own build/test/distribution folders to avoid bloating this repo.

**Comment by Jaifroid** [comment]
Nice summary of a consistent development philosophy, @sharun-s. Note that the Windows Store version is already in a separate repo, kiwix-html5-windows, but I want to keep all the core code in sync as far as is possible. There may have to be some divergences with, for example, setting a higher startup memory footprint for the UWP app.

**Comment by Jaifroid** [comment]
> When there is no other version of Kiwix, I think we could keep a display name "Kiwix" : if there's no other choice, the end-user doesn't care about the underlying technology.

Since there *is* another version of Kiwix which runs on Windows, the Kiwix Win32 binary, I think we can't call the UWP app version "Kiwix". I.e., we already need to distinguish clearly and unambiguously between the two. I think users won't care whether it's called "Kiwix S" or "Kiwix JS". "S" means nothing to me (I don't know what it stands for on iPhones for example -- is it "Special"?), whereas at least "JS" has a meaning someone can work out if they're interested.

CNet says this about the "S" in the iPhone series:

> The "S" in iPhone 4S can stand for anything you want it to, be it Siri or Steve or super. However, after finally spending some time with the iPhone 4S this week, I'm choosing to believe that the S does in fact stand for speed, but you probably already knew that, given the title of this piece.

I think it would look too much like "me too" if we used Apple's "S" (which no doubt they've trademarked in some way as well).

**Comment by kelson42** [comment]
As a result of the discussion I have tried to wrap up and:
* I have renamed the "kiwix-html5" repo to "kiwix-js"
* I have renamed the "kiwix-html5-windows" repo to "kiwix-js-windows"
* I have open a ticket to rebrand "kiwix-html5" as "Kiwix JS" https://github.com/kiwix/kiwix-js/issues/262
* I have open a ticket to rebrand the Kiwix JS version for Windows in "Kiwix JS" https://github.com/kiwix/kiwix-js-windows/issues/8

Thank you all for your respective inputs!
